docs/documentation.rst

```diff
@@ -1,46 +1,119 @@
 
     cd docs
     make html
+    python -m webbrowser file://${PWD}/_build/html/index.html
+
 
-This will generate the documents with the index at
-``docs/_build/html/index.html``.
+To generate the documentation with Docker::
 
+    docker-compose run --rm --user $(id -u) web sh -c "\
+      virtualenv /tmp/.venvs/docs && \
+      . /tmp/.venvs/docs/bin/activate && \
+      pip install -r /app/requirements/docs.txt && \
+      cd /app/docs && \
+      make html"
+    python -m webbrowser file://${PWD}/docs/_build/html/index.html
 
-Updating the MDN Sphinx Theme
------------------------------
-The documentation uses a Sphinx theme generated from the MDN templates:
+A ``virtualenv`` is required, to avoid a ``pip`` bug when changing the version
+of a system-installed package.
 
-https://github.com/mdn/sphinx-theme
+Using the MDN Sphinx Theme project
+----------------------------------
+The documentation uses a Sphinx theme generated from the MDN templates, hosted
+at https://github.com/mdn/sphinx-theme.  This theme is used by MDN and some
+other groups at Mozilla. If you are going to update the theme, you should
+switch from the published version specified in ``docs.txt`` to a local working
+copy.
 
-If this theme is checked out, the in-development theme can be used to generate
-the local documentation::
+When working with a virtualenv on the host machine, the ``sphinx-theme``
+project can be checked out anywhere, and installed in the virtualenv::
 
     pip install -e /path/to/sphinx-theme
-    cd docs
-    make html
 
-The theme's template can be regenerated with::
+Now, when the documentation is built, it will use the current theme in the
+working copy.
+
+Using Docker, the ``sphinx-theme`` project must be checked out inside the Kuma
+repo::
+
+    cd /path/to/kuma
+    git clone git@github.com:mdn/sphinx-theme.git
+
+This will make the sphinx-theme available in Docker at the path
+``/app/sphinx-theme``. When generating documentation, adjust the command to install
+the theme from the working copy::
+
+    docker-compose run --rm --user $(id -u) web sh -c "\
+      virtualenv /tmp/.venvs/docs && \
+      . /tmp/.venvs/docs/bin/activate && \
+      pip install -r /app/requirements/docs.txt && \
+      pip install -e /app/sphinx-theme && \
+      cd /app/docs && \
+      make html"
+    python -m webbrowser file://${PWD}/docs/_build/html/index.html
 
-    ./manage.py generate_sphinx_template > /path/to/sphinx-theme/mdn_theme/mdn/layout.html
+Regenerate the MDN Sphinx Theme template
+----------------------------------------
+When regenerating the template, the Kuma version should be one deployed to
+production.  The assets are loaded from the MDN CDN, and if a non-deployed
+Kuma commit is used to generate the template, files will be unavailable, which
+will may or may not be obvious when you view the documentation locally.
 
+The MDN Sphinx theme template can be regenerated by Kuma::
+
+    docker-compose run --rm -T \
```

Comment:
OK, that's what I expected. It doesn't create the file in the container - it outputs the template, and then it gets redirected to a file as the host user.